### PR TITLE
Intmdb 613 export bucket profile changes

### PR DESCRIPTION
**Pull request by govardhanpagidi**
## Description

CLoud backup export bucket profile changes

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] For CFN Resources: I have released by changes in the private registry and proved by change works in Atlas

### CODE REVIEW

File: cfn-resources/cloud-backup-snapshot-export-bucket/test/add-policy.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
       "Action": "sts:AssumeRole",
       "Condition": {
         "StringEquals": {
-          "sts:ExternalId": "3aa195a5-ecfd-4ed3-9a9f-afe1275067fa"
+          "sts:ExternalId": "6fb95014-fb21-42ed-a2c1-628862460b6a"
```
Comment (colm-quinn) [inline cfn-resources/cloud-backup-snapshot-export-bucket/test/add-policy.json:12]: [q] Where does this value come from? Is it a read resource that must exist in our testing env?

I'm concerned about the test being brittle if the underlying resource is destroyed.

Comment (govardhanpagidi) [reply cfn-resources/cloud-backup-snapshot-export-bucket/test/add-policy.json]: I deleted the values, These are filled by ./test/cfn-test-create-inputs.sh script during test.

File: examples/cloud-backup-snapshot-export-bucket/CloudBackupSnapshotExportBucket.json [hunk 2]
```diff
@@ -1,42 +1,44 @@
...
+        "Profile": {
+          "Ref": "Profile"
         }
       }
     }
   },
-  "Outputs" : {}
-}
\ No newline at end of file
+  "Outputs": {}
```
[the remark this answers is not in the document]
Comment (govardhanpagidi) [reply examples/cloud-backup-snapshot-export-bucket/CloudBackupSnapshotExportBucket.json:43]: @andreaangiolillo  No properties comes from server except the Id, remaining all are sent from client.
I just added Id in output section.

File: cfn-resources/cloud-backup-snapshot-export-bucket/test/cloud-backup-snapshot-export-bucket.sample-cfn-request.json [hunk 3]
```diff
@@ -1,13 +1,10 @@
...
-        "BucketName": "",
-        "ApiKeys": {
-            "PublicKey": "",
-            "PrivateKey": ""
-        }
-    },
-    "providerLogGroupName" : "mongodb-atlas-private-endpoint-logs",
-    "previousResourceState": { }
+  "desiredResourceState": {
+    "GroupId": "",
```
[the remark this answers is not in the document]
Comment (govardhanpagidi) [reply cfn-resources/cloud-backup-snapshot-export-bucket/test/cloud-backup-snapshot-export-bucket.sample-cfn-request.json]: Updated, thanks

File: cfn-resources/cloud-backup-snapshot-export-bucket/cmd/resource/resource.go [hunk 4]
```diff
@@ -60,13 +58,13 @@ func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler
...
 	// Create atlas client
-	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
-	if err != nil {
-		return handler.ProgressEvent{
-			HandlerErrorCode: cloudformation.HandlerErrorCodeInvalidRequest,
-			Message:          err.Error(),
-			OperationStatus:  handler.Failed,
-		}, nil
+	if currentModel.Profile == nil {
+		currentModel.Profile = aws.String(util.DefaultProfile)
```
Comment (andreaangiolillo) [inline cfn-resources/cloud-backup-snapshot-export-bucket/cmd/resource/resource.go:62]:
```suggestion
		currentModel.Profile = aws.String(profile.DefaultProfile)
```

Comment (govardhanpagidi) [reply cfn-resources/cloud-backup-snapshot-export-bucket/cmd/resource/resource.go]: Updated

File: cfn-resources/cloud-backup-snapshot-export-bucket/test/cloud-backup-snapshot-export-bucket.create-sample-cfn-request.sh [hunk 5]
```diff
@@ -19,10 +20,9 @@ groupId="${1}"
...
 
-jq --arg pubkey "$ATLAS_PUBLIC_KEY" \
-	--arg pvtkey "$ATLAS_PRIVATE_KEY" \
-	--arg groupId "$groupId" \
-	--arg iamRoleID "$iamRoleID" \
-	--arg bucketName "$bucketName" \
-	'.desiredResourceState.ApiKeys.PublicKey?|=$pubkey | .desiredResourceState.ApiKeys.PrivateKey?|=$pvtkey | .desiredResourceState.GroupId?|=$groupId | .desiredResourceState.IamRoleID?|=$iamRoleID | .desiredResourceState.BucketName?|=$bucketName ' \
-	"$(dirname "$0")/cloud-backup-snapshot-export-bucket.sample-cfn-request.json"
+jq --arg profile "$ATLAS_PROFILE" \
+   --arg groupId "$groupId" \
```
Comment (andreaangiolillo) [inline cfn-resources/cloud-backup-snapshot-export-bucket/test/cloud-backup-snapshot-export-bucket.create-sample-cfn-request.sh:24]:
```suggestion
   --arg projectId "$projectId" \
```

File: cfn-resources/cloud-backup-snapshot-export-bucket/test/cfn-test-create-inputs.sh [hunk 6]
```diff
@@ -32,59 +33,58 @@ echo "roleName: ${roleName} , policyName: ${policyName}"
...
-echo "--------------------------------authorize mongodb  Role ends ----------------------------"
+echo -e "--------------------------------authorize mongodb  Role ends ----------------------------\n"
 
 #create the s3 bucket
 
-bucketName="cloud-backup-snapshot-test123-"${awsRegion}
+bucketName="cloud-backup-snapshot-${CFN_TEST_TAG}-${awsRegion}"
 
-aws s3 rb "s3://${bucketName}" --force
+#aws s3 rb s3://${bucketName} --force
```
Comment (andreaangiolillo) [inline cfn-resources/cloud-backup-snapshot-export-bucket/test/cfn-test-create-inputs.sh:87]:
```suggestion
```

File: cfn-resources/cloud-backup-snapshot-export-bucket/test/cloud-backup-snapshot-export-bucket.create-sample-cfn-request.sh [hunk 7]
```diff
@@ -19,10 +20,9 @@ groupId="${1}"
...
+   --arg iamRoleID "$iamRoleID" \
+   --arg bucketName "$bucketName" \
+   '.desiredResourceState.Profile?|=$profile | .desiredResourceState.GroupId?|=$groupId | .desiredResourceState.IamRoleID?|=$iamRoleID | .desiredResourceState.BucketName?|=$bucketName ' \
```
Comment (andreaangiolillo) [inline cfn-resources/cloud-backup-snapshot-export-bucket/test/cloud-backup-snapshot-export-bucket.create-sample-cfn-request.sh:27]:
```suggestion
   '.desiredResourceState.Profile?|=$profile | .desiredResourceState.ProjectId?|=$projectId | .desiredResourceState.IamRoleID?|=$iamRoleID | .desiredResourceState.BucketName?|=$bucketName ' \
```

File: cfn-resources/cloud-backup-snapshot-export-bucket/test/cfn-test-create-inputs.sh (hunk 6 above)
Comment (govardhanpagidi) [reply cfn-resources/cloud-backup-snapshot-export-bucket/test/cfn-test-create-inputs.sh:87]: we are creating the bucket freshly everytime, bucket name is randomized

File: cfn-resources/cloud-backup-snapshot-export-bucket/test/cloud-backup-snapshot-export-bucket.create-sample-cfn-request.sh (hunk 7 above)
Comment (govardhanpagidi) [reply cfn-resources/cloud-backup-snapshot-export-bucket/test/cloud-backup-snapshot-export-bucket.create-sample-cfn-request.sh]: @andreaangiolillo  Updated

**Review by andreaangiolillo — APPROVED** [review]

**Pull request merged**